test/graph_generation_tests.cc

```diff
@@ -1459,5 +1459,57 @@ namespace detail {
 		maybe_print_graphs(ctx);
 	}
 
+	TEST_CASE("side effects generate appropriate command-dependencies", "[graph_generator][command-graph][side-effect]") {
+		using order = experimental::side_effect_order;
+		static constexpr auto side_effect_orders = {order::sequential};
```

Comment:
This constant is an input to Catch2's `GENERATE` macro, which basically repeats the test case for each value generated (think dynamically-generated `SECTION`s). Because that iteration can't depend on values local to the test case, `GENERATE` uses a lambda without capture-defaults for the generator expression, so only static variables are permitted.

Reply:
I see. Maybe this is worth a comment explaining the `static`? To me it was absolutely not obvious that it was actually required here.

Reply:
I've added a comment.